pkg/reconciler/workspace/controller.go

```diff
@@ -286,39 +286,67 @@ func (c *Controller) process(ctx context.Context, key string) error {
 }
 
 func (c *Controller) reconcile(ctx context.Context, workspace *tenancyv1alpha1.ClusterWorkspace) error {
-	var shard *tenancyv1alpha1.WorkspaceShard
-	if currentShardName := workspace.Status.Location.Current; currentShardName != "" {
-		// make sure current shard still exists
-		currentShard, err := c.rootWorkspaceShardLister.Get(clusters.ToClusterAwareKey(tenancyhelper.RootCluster, currentShardName))
-		if errors.IsNotFound(err) {
-			klog.Infof("de-scheduling workspace %q|%q from nonexistent shard %q", tenancyhelper.RootCluster, workspace.Name, currentShardName)
-			workspace.Status.Location.Current = ""
-		} else if err != nil {
-			return err
+	switch workspace.Status.Phase {
+	case tenancyv1alpha1.ClusterWorkspacePhaseScheduling:
+		// de-schedule while still in scheduling phase
+		if current := workspace.Status.Location.Current; current != "" {
+			// make sure current shard still exists
+			if _, err := c.rootWorkspaceShardLister.Get(clusters.ToClusterAwareKey(tenancyhelper.RootCluster, current)); errors.IsNotFound(err) {
+				klog.Infof("De-scheduling workspace %q|%q from nonexistent shard %q", tenancyhelper.RootCluster, workspace.Name, current)
+				workspace.Status.Location.Current = ""
+			} else if err != nil {
+				return err
+			}
 		}
-		shard = currentShard
```

Comment:
Why did we lose this?

Reply:
Flow seems correct but is a bit more implicit this way.

Reply:
are you referring to line 375 where we reget the shard?

Reply:
updated. Actually, we only want to touch the baseURL during scheduling. After that it is set in stone forever.